glusterfsd/src/glusterfsd.c

```diff
@@ -597,19 +597,19 @@ create_fuse_mount(glusterfs_ctx_t *ctx)
         }
     }
 
-    ret = xlator_init(master);
+    ret = xlator_init(primary);
     if (ret) {
         gf_msg_debug("glusterfsd", 0, "failed to initialize fuse translator");
         goto err;
     }
 
-    ctx->primary = master;
+    ctx->primary = primary;
```

Comment:
>Just one question. Given that 'primary' is used for geo-replication, wouldn't 'root', 'main' or another term be better to identify this special xlator ?

@xhernandez  I thought about this too, but since I changed `_glusterfs_ctx->master` to `_glusterfs_ctx->primary` in an earlier commit (e4c9a14429c51d8d059287c2a2c7a76a5116a362), I thought this particular line would look odd (i.e. ctx->primary = root etc). Do you think its better to change that to root too? I can refresh this patch in that case.

Reply:
ctx->root is a good idea. Considering ctx->primary already exists, lets take this PR in, and you can send one more to change all the primary (here) to root.

Reply:
@amarts Ack!

Reply:
> @xhernandez I thought about this too, but since I changed `_glusterfs_ctx->master` to `_glusterfs_ctx->primary` in an earlier commit ([e4c9a14](https://github.com/gluster/glusterfs/commit/e4c9a14429c51d8d059287c2a2c7a76a5116a362)), I thought this particular line would look odd (i.e. ctx->primary = root etc). Do you think its better to change that to root too? I can refresh this patch in that case.

You are right. I missed this detail when I reviewed the previous patch. As @amarts says, we can go with this patch and change it later if needed.